invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/UppyUploader/InvenioMultipartUploader.js

```diff
@@ -0,0 +1,384 @@
+// This file is part of Invenio-RDM-Records
+// Copyright (C) 2020-2025 CERN.
+// Copyright (C)      2025 CESNET.
+//
+// Invenio-RDM-Records is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+// NOTE: If WASM undesirable, we can use library like Crypto-JS
+import { md5 } from "hash-wasm";
+import AwsS3Multipart from "@uppy/aws-s3-multipart";
+import { fetcher } from "@uppy/utils/lib/fetcher";
+import { humanReadableBytes } from "react-invenio-forms";
+
+import { FileSizeError, InvalidPartNumberError, SignedUrlExpiredError } from "./error";
+
+const defaultOptions = {
+  // NOTE: null here means “include all”, [] means include none.
+  // This is inconsistent with @uppy/aws-s3 and @uppy/transloadit
+  allowedMetaFields: null,
+  limit: 3,
+  getTemporarySecurityCredentials: false,
+  getUploadParameters: null,
+  shouldUseMultipart: null,
+  uploadPartBytes: null,
+  retryDelays: [0, 1000, 3000, 5000],
+  companionHeaders: {},
+  checkPartIntegrity: false,
+};
+
+export class InvenioMultipartUploader extends AwsS3Multipart {
+  static VERSION = "1.0.0";
+  #maxMultipartParts = 10_000;
+
+  constructor(uppy, opts) {
+    super(uppy, {
+      ...defaultOptions,
+      fieldName: opts.bundle ? "files[]" : "file",
+      ...opts,
+    });
+    this.type = "uploader";
+    this.draftRecord = this.opts.draftRecord;
+    this.id = this.opts.id || "InvenioMultipartUploader";
+    this.getChunkSize = this.opts.getChunkSize || this.getChunkSize;
+    this.shouldUseMultipart = this.opts.shouldUseMultipart || this.shouldUseMultipart;
+
+    this.i18nInit();
+    super.setOptions({
+      // Here we override default implementation in AwsS3Multipart
+      shouldUseMultipart: this.shouldUseMultipart.bind(this),
```

Comment:
given that this class firstly implements an Uppy file uploader interface and additionally adds the multipart functionality, does it make sense to remove the multipart from the class name? It could be renamed to `RDMUppyUploaderPlugin` (I see it is added as plugin to the Uppy instance) and the docstrings are explaining sufficiently what is the algorithm behind using multipart or single upload. What do you think?

Reply:
Good point, current naming is confusing, as this Uppy uploader plugin actually handles both upload types (single & mp). I'll rename